[Backtracing] Add missing availability annotations.

al45tair opened the pull request:
This is slightly complicated by us currently not supporting this code on iOS/tvOS/watchOS/visionOS.  We should fix that, at which point we can use the `SwiftStdlib`/`StdlibDeploymentTarget` macros instead.

rdar://164850733

al45tair committed:
[Backtracing] Add missing availability annotations.

This is slightly complicated by us currently not supporting this code
on iOS/tvOS/watchOS/visionOS.  We should fix that, at which point we
can use the `SwiftStdlib`/`StdlibDeploymentTarget` macros instead.

rdar://164850733

compnerd commented on Runtimes/Supplemental/Runtime/CMakeLists.txt:
```diff
@@ -93,7 +93,8 @@ add_compile_options(
   "$<$<AND:$<BOOL:${${PROJECT_NAME}_ENABLE_LIBRARY_EVOLUTION}>,$<COMPILE_LANGUAGE:Swift>>:-enable-library-evolution>"
-  "$<$<AND:$<BOOL:${${PROJECT_NAME}_ENABLE_PRESPECIALIZATION}>,$<COMPILE_LANGUAGE:Swift>>:SHELL:-Xfrontend -prespecialize-generic-metadata>")
+  "$<$<AND:$<BOOL:${${PROJECT_NAME}_ENABLE_PRESPECIALIZATION}>,$<COMPILE_LANGUAGE:Swift>>:SHELL:-Xfrontend -prespecialize-generic-metadata>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-Xfrontend -define-availability -Xfrontend \"Backtracing 6.2: macOS 26.0\">")
```
This should not be added here. We auto-generate the availability macros in https://github.com/al45tair/swift/blob/c74154533b3e648a8f8e31f4e655e77da0042eb0/Runtimes/Supplemental/cmake/modules/AvailabilityMacros.cmake

Please add this to utils/availability-macros.def instead

al45tair replied:
Not this one we don't, by design.

al45tair replied:
See the code in the other `CMakeLists.txt` for why.

compnerd reviewed (changes requested):
This shouldn

compnerd reviewed (changes requested):
responded in thread

al45tair committed:
[Backtracing] More availability annotations required.

Apparently we need some more availability annotations.

rdar://164850733

al45tair committed:
[Backtracing] Move availability definition, fix tests.

Move the availability definition to the `.def` file, and then
turn off availability checking in the tests.

rdar://164850733

al45tair committed:
[Tests] Fix lit.cfg to account for the new availability definition.

`availability_macro_to_swift_abi_target_triple` failed when it saw
our new availability macro.

rdar://164850733

al45tair commented:
@swift-ci Please test

edymtt reviewed (approved):
Looks good to me

For future reference, to be able to use `SwiftStdlib`/`StdlibDeploymentTarget` availabilities we would need to ensure to use the same logic present in `Runtimes/Core/cmake/modules/AvailabilityMacros.cmake` to build the backtracing module in the new build system -- either by generalizing that and moving that in `Runtimes/cmake/modules`, or by copying its implementation in `Runtimes/Supplemental/cmake/modules/AvailabilityMacros.cmake`.

al45tair commented:
@swift-ci Please smoke test

al45tair committed:
[Backtracing][Tests] `-disable-availability-checking` is frontend-only.

We need to add `-Xfrontend`.

rdar://164850733

al45tair commented:
@swift-ci Please smoke test

al45tair committed:
[Backtracing] Add some more availability annotations.

We needed a few more annotations, it turns out.  Also, we need to disable
availability checking when building swift-backtrace.

rdar://164850733
